Symbolic Navier-Stokes element test

rubenzorrilla opened the pull request:
Symbolic Navier-Stokes element LHS and RHS test (together with minor changes in the symbolic generation).

rubenzorrilla committed:
Merged master into feature-navier-stokes-element-test

rubenzorrilla committed:
Addition of NavierStokes symbolic element tests

New .cpp tests for NavierStokes2D3N and NavierStokes3D4N elements.

rubenzorrilla committed:
Modifications and advances in the test structure

rubenzorrilla committed:
Merged branch master into feature-navier-stokes-element-test

rubenzorrilla committed:
Merge remote-tracking branch 'refs/remotes/origin/feature-navier-stokes-element-test'

Conflicts:
	applications/FluidDynamicsApplication/tests/test_navier_stokes_symbolic_element.cpp

rubenzorrilla committed:
Merge remote-tracking branch 'refs/remotes/origin/feature-navier-stokes-element-test'

Conflicts:
	applications/FluidDynamicsApplication/tests/test_navier_stokes_symbolic_element.cpp

rubenzorrilla committed:
Merge remote-tracking branch 'refs/remotes/origin/feature-navier-stokes-element-test'

Conflicts:
	applications/FluidDynamicsApplication/tests/test_navier_stokes_symbolic_element.cpp

rubenzorrilla committed:
Merge remote-tracking branch 'refs/remotes/origin/feature-navier-stokes-element-test'

Conflicts:
	applications/FluidDynamicsApplication/tests/test_navier_stokes_symbolic_element.cpp

rubenzorrilla committed:
Merge remote-tracking branch 'refs/remotes/origin/feature-navier-stokes-element-test'

Conflicts:
	applications/FluidDynamicsApplication/tests/test_navier_stokes_symbolic_element.cpp

rubenzorrilla committed:
Improvements in symbolic derivation

Minor changes in the symbolic derivation script. Now the stabilization parameters are included in the derivation. The default iteration is set to be the Picard's one.

rubenzorrilla committed:
Tests modifications

Modifications in the Navier-Stokes symbolic element according to the last implementations. Addition of the updated LHS and RHS test.

RiccardoRossi commented on applications/FluidDynamicsApplication/CMakeLists.txt:
```diff
@@ -37,6 +36,8 @@ set( KRATOS_FLUID_DYNAMICS_APPLICATION_SOURCES
   ${CMAKE_CURRENT_SOURCE_DIR}/custom_constitutive/newtonian_2d_law.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/custom_constitutive/newtonian_3d_law.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/custom_constitutive/herschel_bulkey_3d_law.cpp
+
```
i think this should be in only if tests are compiled

rubenzorrilla replied:
Totally agree. I will do it asap. Thanks @RiccardoRossi .

RiccardoRossi reviewed (approved):
just ensure that compilation is conditional (if tests are applied). other than that go ahead

loumalouomega commented:
![](https://media.giphy.com/media/aLdiZJmmx4OVW/giphy.gif)

NICE WORK

rubenzorrilla committed:
Optional test compilation

Now the cpp fluid dynamic tests will be compiled if DKRATOS_BUILD_TESTING matches ON.

rubenzorrilla reviewed (commented):
@roigcarlo I think that this is ok, but would you mind to have a look? Thanks!

roigcarlo commented:
Yup that's it :+1:
